system/jlib/jsocket.cpp

```diff
@@ -4826,14 +4826,18 @@ class CSocketEpollThread: public CSocketBaseThread
                                     if (ep_mode != 0)
                                     {
                                         tonotify.append(*epsi);
+                                        SelectItem &itm = tonotify.element(tonotify.length()-1);
+                                        itm.nfy->Link();
+                                        itm.sock->Link();
```

Comment:
As far as I can see, CSocketSelectThread suffers from the same problem, so similar change should be made there.